### PR TITLE
Allow users to specifiy their own custom locale-independent dateformat

### PR DESCRIPTION
**Pull request by wolfposd**
- added custom dateformat setting to conf

### CODE REVIEW

**Comment by tintou** [comment]
The introduction of `ISO8601` as a "clock format" was already mistake, what you want is a datetime format which should be a different key.

**Comment by wolfposd** [comment]
should i create a date format, and use it analogue to the clock-format?

**Comment by jeremypw** [comment]
This works but some validation of user input should be carried out.  If the setting is invalid it should revert to a safe value.

**Comment by jeremypw** [comment]
Maybe the setting description should give more info on how to format the string.

File: src/Widgets/PanelLabel.vala [hunk 1]
```diff
@@ -55,14 +60,19 @@ public class DateTime.Widgets.PanelLabel : Gtk.Grid {
...
 
     private void update_labels () {
-        string date_format;
-        if (clock_format == "ISO8601") {
-            date_format = "%F";
-        } else {
-            date_format = Granite.DateTime.get_default_date_format (clock_show_weekday, true, false);
+        string date_formatString;
+        
+        if(date_format == "custom") {
```
Comment (jeremypw) [inline src/Widgets/PanelLabel.vala]: Missing space before '('

File: src/Widgets/PanelLabel.vala [hunk 2]
```diff
@@ -55,14 +60,19 @@ public class DateTime.Widgets.PanelLabel : Gtk.Grid {
...
+        	date_formatString = date_format_custom;
+        }
+        else if (date_format == "ISO8601") {
```
Comment (jeremypw) [inline src/Widgets/PanelLabel.vala]: Does not follow elementaryos code style: use ' } else if (...) {'

File: src/Widgets/PanelLabel.vala [hunk 3]
```diff
@@ -55,14 +60,19 @@ public class DateTime.Widgets.PanelLabel : Gtk.Grid {
...
+        	date_formatString = "%F";
         }
+        else {
```
Comment (jeremypw) [inline src/Widgets/PanelLabel.vala]: Code style - see above

**Comment by danrabbit** [comment]
Gonna close this since there hasn't been any movement in months and this conflicts with master.

If we want to add more clock format options, we now need to first make them available in the pantheon accountsservice extension provided by `default-settings`

**Pull request closed without merging**